lib/src/main/kotlin/com/swmansion/starknet/data/types/Transaction.kt

```diff
@@ -160,7 +161,28 @@ data class DeployTransaction(
 
     @SerialName("type")
     override val type: TransactionType = TransactionType.DEPLOY,
-) : Transaction(), DeprecatedTransaction
+) : Transaction(), DeprecatedTransaction {
+    constructor(
```

Comment:
I think we should use `@JvmOverloads` for the secondary constructor here as well, as adding a secondary constructor (I presume to hardcode fields like `type = TransactionType.DEPLOY`), doesn't solve the **defaults**